client/components/theme-preview-modal/index.tsx

```diff
@@ -96,6 +98,10 @@ const ThemePreviewModal: React.FC< ThemePreviewModalProps > = ( {
 		};
 	}, [] );
 
+	useEffect( () => {
+		onShouldLimitGlobalStyles?.( shouldLimitGlobalStyles );
```

Comment:
Wondering what is it for? I tried removing this `useEffect` block locally and I've found no issues.

Reply:
This is to notify the parent component wherever the `shouldLimitGlobalStyles` value changes. By default, it's set to `false` since it's pre API fetch. If the API returns `true`, we call the callback `onShouldLimitGlobalStyles` to let the parent component know that we should limit global styles. In this use case, we change the action button to "Unlock this style" when limiting global styles.

Let me know if that answers your question, and suggestions welcomed 🙂

Reply:
I see, I find it a bit hard to understand because a parent component's behavior is dictated from API call done from a child component.

I was going to suggest that we pass the `shouldLimitGlobalStyles` prop from `ThemePreview` to `ThemePreviewModal`, but we can't since `ThemePreview` cannot use the `useSiteGlobalStylesStatus` hook because it's a class component 🤦 

Either we leave it as is for now, or refactor `ThemePreview` to be functional component 😄 your decision!

Reply:
> I think it's a bit high-risk to refactor ThemePreview.

Makes sense due to this reason 😄

Reply:
I think we don't need to refactor `ThemePreivew` component. Instead, we can replace its HOC such as connect with a functional component and inject the props what we want.

For example

```js
const ThemePreviewContainer = ( props ) => {
  const themeId = useSelector( themePreviewVisibility );
  const siteId = useSelector( getSelectedSiteId );

  ...

  const dispatch = useDispatch();

  return (
    <ThemePreview
      {...props}
      themeId={ themeId }
      siteId={ siteId }
      ...
      hideThemePreview={ () => dispatch( hideThemePreview() ) }
      ...
    />
  );
}
```

What do you think?

Reply:
I've refactored the component by using a HOC to use `useSiteGlobalStylesStatus()` in `<ThemePreview/>` 🙂